ramalama/cli.py

```diff
@@ -857,14 +857,14 @@ def rm_cli(args):
 
 
 def New(model, args):
-    if model.startswith("huggingface://") or model.startswith("hf://") or model.startswith("hf.co/"):
+    if model.startswith("hf://") or model.startswith("huggingface://") or model.startswith("hf.co/"):
         return Huggingface(model)
-    if model.startswith("ollama"):
-        return Ollama(model)
-    if model.startswith("oci://") or model.startswith("docker://"):
-        return OCI(model, args.engine)
-    if model.startswith("http://") or model.startswith("https://") or model.startswith("file://"):
+    elif (
+        model.startswith("https://") or model.startswith("http://") or model.startswith("file://")
+    ) and not model.startswith("https://ollama.com/library/"):
         return URL(model)
+    elif model.startswith("oci://") or model.startswith("docker://"):
+        return OCI(model, args.engine)
```

Comment:
My question is for this one, can we have util.py this seems very useful as generic.

Reply:
I'm not against more refactoring in future PRs. What part exactly would go to utils this whole function?